### PR TITLE
refactor(semver): make `outside()` more readable

### PR DESCRIPTION
**Pull request by timreichen**
- makes `outside()` more readable by replacing the algorithm with already existing functions.

As pointed out in https://github.com/denoland/deno_std/issues/3948#issuecomment-1876875415, this seems now to be a weird function to include in the semver mod.

### CODE REVIEW

**Review by iuioiua — COMMENTED** [review]
Are you able to add the test case mentioned in https://github.com/denoland/deno_std/issues/3948#issuecomment-1876875415 ?

**Comment by timreichen** [comment]
> Are you able to add the test case mentioned in [#3948 (comment)](https://github.com/denoland/deno_std/issues/3948#issuecomment-1876875415) ?

Done.

File: cli/parse_args.ts
Comment (iuioiua) [inline cli/parse_args.ts:1]: I suspect these changes have been mistakenly added.

Comment (timreichen) [reply cli/parse_args.ts:1]: Sorry, that wasn't supposed to be in that PR. Reverted

**Comment by timreichen** [comment]
Actually, would it make sense to additionally deprecate `outside()` and link to `!testRange()`, `gtr` and `ltr`? These are the three use cases `outside()` handles.

**Review by iuioiua — APPROVED** [review]
LGTM, as this removes the dependency on `outside()`, which _might_ be worth deprecating.

Please see the comment on the new test that I've added. We should relocate, then tweak this test if the reported behaviour is agreed upon to be a bug (which both Tim and I already do).

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**